source/opt/module.cpp

```diff
@@ -175,13 +177,28 @@ void Module::ToBinary(std::vector<uint32_t>* binary, bool skip_nop) const {
         last_line_inst = nullptr;
       }
     }
+
+    if (opcode == SpvOpLabel) {
+      between_label_and_phi_var = true;
+    } else if (opcode != SpvOpVariable && opcode != SpvOpPhi &&
+               opcode != SpvOpLine && opcode != SpvOpNoLine) {
+      between_label_and_phi_var = false;
+    }
+
     if (!(skip_nop && i->IsNop())) {
       const auto& scope = i->GetDebugScope();
       if (scope != last_scope) {
-        // Emit DebugScope |scope| to |binary|.
-        auto dbg_inst = ext_inst_debuginfo_.begin();
-        scope.ToBinary(dbg_inst->type_id(), context()->TakeNextId(),
-                       dbg_inst->GetSingleWordOperand(2), binary);
+        // don't emit scope changes when outside of blocks in
+        // Vulkan.NonSemantic.DebugInfo.100
```

Comment:
I cannot understand this comment. Could you please rephrase it?
In addition, please start the sentence with a large character instead of a small one e.g., `Don't ..` instead of `don't`
and end the sentence with a dot `.`.

Reply:
Nonsemantic instructions cannot be emitted before any phi instructions in a block. This logic will make sure scope instructions are emitted after any phi instructions in a block for Vulkan.NonSemantic.DebugInfo.100. I have updated the comment to make this clearer.